Remotion/Data/DomainObjects.UberProfIntegration/EntityFrameworkAppenderProxy.cs

```diff
@@ -0,0 +1,174 @@
+// This file is part of the re-motion Core Framework (www.re-motion.org)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+// 
+// The re-motion Core Framework is free software; you can redistribute it 
+// and/or modify it under the terms of the GNU Lesser General Public License 
+// as published by the Free Software Foundation; either version 2.1 of the 
+// License, or (at your option) any later version.
+// 
+// re-motion is distributed in the hope that it will be useful, 
+// but WITHOUT ANY WARRANTY; without even the implied warranty of 
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the 
+// GNU Lesser General Public License for more details.
+// 
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+// 
+using System;
+using System.Data;
+using System.Reflection;
+using System.Runtime.Serialization;
+using Remotion.Utilities;
+
+namespace Remotion.Data.DomainObjects.UberProfIntegration
+{
+  /// <summary>
+  /// Implements a strong-typed wrapper for <b><a href="https://hibernatingrhinos.com/products/efprof">Entity Framework Profiler</a></b>.
+  /// </summary>
+  /// <remarks>
+  /// The wrapper uses runtime-binding to redirect the calls to Entity Framework Profiler's API. This removes the static dependency on Entity Framework Profiler.
+  /// </remarks>
+  /// <threadsafety static="true" instance="true" />
+  [Serializable]
+  public sealed class EntityFrameworkAppenderProxy : IObjectReference
+  {
+    private static readonly DoubleCheckedLockingContainer<EntityFrameworkAppenderProxy> s_instance =
+        new DoubleCheckedLockingContainer<EntityFrameworkAppenderProxy>(
```

Comment:
Since it's new code: please switch to Lazy. The public setter can be dropped. Don't know why it's on LinqToSql Proxy, but the ctor is private so why would you ever...? So, anyway, please make the new code sane from the get go :)

Reply:
Problem is that the variable is set via reflection from the tests. Changing that to a `Lazy<T>` is more work since it is not easily settable via reflection. So I would keep the code as is.